### PR TITLE
Bindings Archetype: Move Handler and Constants into internal package

### LINKED ISSUE #4330: Exported handlers
Let's continue this one: https://github.com/eclipse/smarthome/pull/4311#issuecomment-331152732

If handlers are exported the API of the handler implementations must not use private packages ("must" if we don't like to create a broken API).

See e.g. http://enroute.osgi.org/tutorial_wrap/270-private-references.html

So, we should consider if we want exported handlers and fix the existing ones (some time) or if we should change the archetype and existing bindings.

### PR DESCRIPTION
**Pull request by triller-telekom**
As discussed in #4330 it makes sense to export handlers to create a derived
handler in another bundle. We do have this for the bluetooth binding and
will have it for the zigbee binding as well.

These two bindings are the rare exception and their handler APIs have been
throught through, i.e. they have a stable API and do not expose internal
classes.

Thus for newly created bindings it makes sense to NOT export the handlers
and if there should be a derived handler at some point in the future we
know the use-cases for the derived handler and can design a proper API
for the handler at that time and then export the handler.

Fixes #4330

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/handler/__bindingIdCamelCase__Handler.java [hunk 1]
```diff
@@ -17,9 +17,9 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  */
-package ${package}.handler;
+package ${package}.internal.handler;
```
[the remark this answers is not in the document]
Comment (triller-telekom) [reply tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/handler/__bindingIdCamelCase__Handler.java]: It was in a `handler` package before and thus I moved it with its package.

Also when reviewing a bundle it helps to have an overview about which handlers exist if they are all in one package.

I would vote for keeping them in an extra package. But if others insist of moving it to the `internal` package without the extra `handler` part, I'll do it.

**Comment by triller-telekom** [comment]
I have removed the leftover imports. And to answer your question: Yes I have tried in in my IDE and it did **not** complain about that import although it was't exported anymore.

File: tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/handler/__bindingIdCamelCase__Handler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaikreuzer) [reply tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/handler/__bindingIdCamelCase__Handler.java]: Packages should be designed in a way that each represents a cohesive set of classes, defining an API. Packages should NOT be used to structure classes by type.

[the remark this answers is not in the document]
Comment (triller-telekom) [reply tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/handler/__bindingIdCamelCase__Handler.java]: Grouping them by API makes sense. I have moved it in my last commit.

**Pull request merged**